### ISSUE TITLE
Stack View - Feature Request

### ISSUE THREAD

**Issue by callen5914** [vmware/clarity#2857]
## Summary

Can we have [clrSbSelectable]="true : false"?
Can we have [clrSbSelected]="true"?

## Use case

I think the DataGrid is brilliant but in some cases overkill. The Stack View is a great UI component is Tree View and/or a standard list doesn't quite fit into the UI.

If we had the ability to have a selectable Stack Node even if it doesn't have children and the CSS styles needed to signify a selection, this could be of benefit. For instance, if one was selected; the selected node could have data displayed in a card to the right of the Stack.

## Examples

Unfortunately, No. The closest thing I could find was Clarity's very own Data grid. I would think this could be dressed down a little bit.

## Workarounds

I tried to add:

```
<clr-stack-block *ngFor="let base of bases" (click)="onSelect(base)" [class.selected]="base === selectedBase">
    <clr-stack-label>{{ base.baseName }}</clr-stack-label>
    <clr-stack-content>{{ base.isActive? 'Active' : 'Inactive' }}</clr-stack-content>
</clr-stack-block>

```

Then of course use CSS to tap into the node and highlight the selected node. But it's not really happening. I think it has something to do with how it gets rendered.

Cheers,

Coach

**Comment by youdz** [comment]
For both accessibility and HTML-validity reasons, the `<clr-stack-block>` element would not be clickable. We would also need to make sure we display a checkbox or radio button on each block (and potentially each child), for the same reasons we have them in the datagrid.
On top of that, there would simply be no way for you select a parent by clicking on its label, because it would conflict with the expand/collapse behavior. Same thing for selecting by clicking on the content, it would conflict with any interactive elements in there (which we don't control).

In other words, I can see many constraints for accessibility and reusability that mean the design for this would probably not look and behave like what you're expecting. We can take a look at this to see if there's a design for selectable stack views that we want to support.

Personally, I highly recommend you use a Tree View for this, simply customizing the styles of the nodes if you feel it doesn't match your use case.

**Comment by gnomeontherun** [comment]
Based on the above, closing this as unactionable. We can't have conflicting behaviors for expand/selection, and alternatives exist.
